maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/booterclient/ForkStarter.java

```diff
@@ -656,6 +652,10 @@ private RunResult fork( Object testSet, PropertiesWrapper providerProperties, Fo
                 errConsumer, countdownCloseable );
             err.start();
 
+            log.debug( "Waiting for fork [" + forkNumber + "] process to connect..." );
```

Comment:
You cannot move this part, you cannot read data unless you are connected to the client. If you expect any error message, then it has to come from the `accept()` method and the `get()` method.

Reply:
Well, I do not know why those builds pass... but in case my case it definitely fixes the locking forked process when having an output on the forked process when adding the `-verbose:classes` JVM argument...